### PR TITLE
Hide input controls while chat is streaming

### PR DESCRIPTION
**Pull request by malmstein**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1214157224317277/task/1214882460531879

### Description
While a Duck.ai chat is streaming, the bottom-row controls (model picker, reasoning picker, options, attach button) are hidden and the stop button moves inline with the text field (row 1). When streaming ends the controls restore correctly.

Root cause: `updateBottomRowVisibility` kept the entire bottom row visible during streaming so the stop button would show — causing the four controls to flash visible too. Fix:
- Extracts a pure `shouldShowInputControls(onChatTab, isStreaming)` predicate (unit-tested)
- Adds a dedicated `streamingButtonsContainer` in the input field row for the stop button during streaming
- Hides the bottom row entirely while streaming (`!isStreaming` gate in `updateBottomRowVisibility`)
- Respects tool-picker state (image generation) when restoring controls after streaming ends

### Steps to test this PR

_Hide controls while streaming_
- [ ] Settings → AI Features → "Search Only"
- [ ] Focus the omnibar → tap the Duck.ai button to start a chat
- [ ] While streaming: only the stop button is visible, inline with the text field — model picker, reasoning picker, options and attach button are hidden, controls row is not visible
- [ ] After streaming ends: controls row reappears on focus with send button, model picker, reasoning picker, options and attach button visible as normal
- [ ] Sanity: in a focused Duck.ai input not streaming, all controls are visible as before

### UI changes
| Before  | After |
| ------ | ----- |
|Upload before screenshot|Upload after screenshot|

---

> [!NOTE]
> **Medium Risk**
> Touches omnibar/native input visibility and stop-stream wiring across multiple plugin views; behavior is covered by new unit tests but UX regressions on tab switch or streaming edge cases are possible.
> 
> **Overview**
> While Duck.ai is **streaming or loading**, the native input chrome no longer keeps the bottom tool row visible for a stop affordance. **`isChatStreaming`** is added to **`NativeInputState`**, driven from **`ChatState.STREAMING` / `LOADING`** in the widget ViewModel and published per tab.
> 
> **Stop** moves **inline** with the text field via a new **`StopStreamingNativeInputPlugin`** (`streamingButtonsContainer` + **`NativeInputHost.stop()`**). The bottom row is **fully hidden** during streaming (`updateBottomRowVisibility` gates on `!isStreaming`), and attach / model / reasoning / options plugins **self-hide** using **`shouldShowPluginControls()`** (Duck.ai tab + not streaming) instead of the removed **`UpdatePluginVisibility`** command channel and tab-driven container toggles.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 59141e8b00231d0f3ee9a771cb53578f28f9ee7e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

**Review by joshliebe — APPROVED** [review]
LGTM

File: duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/ui/nativeinput/views/NativeInputModeWidget.kt [hunk 1]
```diff
@@ -457,11 +459,26 @@ class NativeInputModeWidget @JvmOverloads constructor(
...
+        // The model/reasoning pickers are additionally owned by OptionsView, which hides them for
+        // tools that don't use them (e.g. image generation). Respect that so restoring controls
+        // after streaming doesn't re-show pickers that should stay hidden.
+        val showPickers = show && (optionsView?.pickersEnabled ?: true)
+        findViewById<View?>(R.id.modelPickerContainer)?.isVisible = showPickers
+        findViewById<View?>(R.id.reasoningModePickerContainer)?.isVisible = showPickers
+        findViewById<View?>(R.id.optionsButtonContainer)?.isVisible = show
+        findViewById<View?>(R.id.attachmentsContainer)?.isVisible = show && hasAttachments
+        // attach button keeps its own supportsUpload gate; setImageButtonVisible folds in !isStreaming
+        setImageButtonVisible(isChatTabSelected() && supportsUpload)
```
Comment (joshliebe) [inline duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/ui/nativeinput/views/NativeInputModeWidget.kt]: This logic should probably be controlled by the viewmodel

Comment (malmstein) [reply duckchat/duckchat-impl/src/main/java/com/duckduckgo/duckchat/impl/ui/nativeinput/views/NativeInputModeWidget.kt]: sorted
